Neuro/Minigames/MinigameTimeHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Il2CppInterop.Runtime.Attributes;
+using System;
+using Neuro.Events;
+using Neuro.Utilities;
+using Reactor.Utilities.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Neuro.Minigames;
+
+[RegisterInIl2Cpp, ShipStatusComponent]
+public sealed class MinigameTimeHandler : MonoBehaviour
+{
+    public static MinigameTimeHandler Instance { get; private set; }
+
+    [HideFromIl2Cpp]
+    public Dictionary<string, List<float>> MinigameTimes { get; } = new Dictionary<string, List<float>>();
+
+    private string _minigameName;
+    private float _startTime;
+    private Func<bool> _stopTimerCondition;
+
+    public MinigameTimeHandler(IntPtr ptr) : base(ptr)
+    {
+    }
+
+    private void Awake()
+    {
+        if (Instance)
+        {
+            NeuroUtilities.WarnDoubleSingletonInstance();
+            Destroy(this);
+            return;
+        }
+
+        Instance = this;
+    }
+
+    private void Update()
+    {
+        if (_stopTimerCondition != null && _stopTimerCondition())
+        {
+            Stop();
+        }
+    }
+
+    private void AddMinigameTime(float time)
+    {
+        if (MinigameTimes.TryGetValue(_minigameName, out List<float> list))
+        {
+            if (list.Last() > 0.0f)
+            {
+                MinigameTimes[_minigameName].Add(time);
+            }
+        }
+        else
+        {
+            MinigameTimes[_minigameName] = new List<float>() { time };
+        }
+    }
+
+    public void Clear()
+    {
+        MinigameTimes.Clear();
+    }
+
+    public void StartTimer(Minigame minigame, Func<bool> hideCondition)
+    {
+        _minigameName = minigame.GetIl2CppType().Name;
```

Comment:
This might be a problem because some minigames (such as the water can minigame on mira) use the same minigame class for both steps. In this case it's not very important as the times are very similar, but it might be worth storing the task type + task step / task timer state alongside the minigame name.

Reply:
So we want the task type + step to differentiate between them instead? We would get essentially duplicates that way, with tasks like wiring that have multiple parts of the same thing. Up to you on which direction is more preferable.

Reply:
Pushed up a commit that should address this desire, but would like a look over to make sure it's what's expected.